src/framework/actions/hand.ts

```diff
@@ -4,6 +4,96 @@ import { GameState } from "../types";
 
 import { makeInstance } from "./helpers";
 
+/**
+ * NOTE: For security purposes, drawing should be handled or confirmed by the
+ * server. Active players' clients should not be allowed to manipulate their
+ * draw order, and the contents/order of the draw pile for other players needs
+ * to be unknown.
+ * NOTE: This draws the top card. For abilities that allow drawing an arbitrary
+ * card from the player's deck, a different function should be used.
+ *
+ * @param $
+ * @param pid
+ */
+export const drawCard = ($: GameState): void => {
+  const P = $.players[$.activePlayer];
+
+  if (P.deck.length > 0) {
+    P.hand = P.hand.concat(P.deck.splice(0, 1));
+  } else if (P.hasShuffledThisTurn === false) {
+    P.hasShuffledThisTurn = true;
+    shuffle($);
+    drawCard($);
+  } else {
+    throw new Error("deck is empty");
```

Comment:
From the rules:

>  If you manage to empty your draw pile a second time during your main phase, you can’t draw any more cards during that main phase

The rules don't make it explicitly clear what should happen if you are unable to draw a card, but my assumption would be that you simply don't draw anything (as opposed to it being illegal to do something that would cause you to need to draw a card when your deck is empty). An error here should essentially mean "whatever action you just tried to do, _you can't_"; i.e., this would mean if someone tries to kill an enemy technician when they have no cards in deck, their attack is actually not allowed.

With that in mind, I think we should change this to return `boolean` and have it return whether a card was successfully drawn or not (in case there's a card like "draw a card; if you do, ___"), but this should never error.

Reply:
I intended the behavior that the rules describe, but for some reason I didn't consider how counter-productive an exception would be (more burden on the caller).

I've made this correction now.